stl/inc/flat_set

```diff
@@ -420,7 +428,7 @@ public:
         return _RANGES equal(_Lhs._Get_cont(), _Rhs._Get_cont());
     }
 
-    _NODISCARD friend _Synth_three_way_result<_Kty> operator<=>(const _Deriv& _Lhs, const _Deriv& _Rhs) {
+    _NODISCARD friend auto operator<=>(const _Deriv& _Lhs, const _Deriv& _Rhs) {

```

Comment:
This was making `test_comparer_application` fail to compile.

Reply:
Maybe not a compiler bug, the code is rejected by both msvc, gcc and clang :|
https://godbolt.org/z/jT6sT4EfT (the error message given by msvc is the strangest one though)
```cpp
#include<utility>

#define USE_FRIEND

struct myless {
    template<class L, class R>
    auto operator()(const L& lhs, const R& rhs) {
        return lhs < rhs;
    }
};

template<class L, class R = L>
using myless_result = decltype(myless{}(std::declval<L&>(), std::declval<R&>()));

template<class T>
struct wrapper {
    T t;
#if defined USE_FRIEND
#if 1
    // "unexpected token(s) preceding '{'; skipping apparent function body"
    // then other errors.
    friend myless_result<T> operator<(const wrapper& lhs, const wrapper& rhs) {
        return myless{}(lhs.t, rhs.t);
    }
#else
    // will be ok if returns auto.
    friend auto operator<(const wrapper& lhs, const wrapper& rhs) {
        return myless{}(lhs.t, rhs.t);
    }
#endif

#endif // USE_FRIEND
};

#if !defined USE_FRIEND
// will be ok if not friend.
template<class T>
myless_result<T> operator<(const wrapper<T>& lhs, const wrapper<T>& rhs) {
    return myless{}(lhs, rhs);
}
#endif // !USE_FRIEND

struct xcmp {
    bool operator=(const xcmp&)const = delete;
    bool operator<(const xcmp&)const = delete;
};

int main() {
    wrapper<xcmp> test;
}
```